### PR TITLE
Fixed selling bug

### PR DESCRIPTION
**Pull request by Fueredoriku**
Fixed a bug where selling a tower after dying would give errors.

accidentally managed to push to develop, so here is the original commit message:

> Made refunding turrets it's own function, and updated the towerScriptableObject with get function. While testing for this bugfix, I found that playerstatecontroller doesn't actually set or update liftedObject, which made the scriptableobject unaccessable. Until this is resolved, the sellTower function in playerstatecontroller will use the holo prefab for cost variable access.

### CODE REVIEW

**Review by ddabble — CHANGES_REQUESTED** [review]
Not sure why you removed so much whitespace 🤔 Would you mind reverting those changes, to keep the code style consistent with the rest of the codebase? 🙂

File: Assets/Scripts/Player/PlayerStateController.cs [hunk 1]
```diff
@@ -30,38 +30,38 @@ public partial class PlayerStateController : MonoBehaviour
     private InventoryManager inventoryManager;
 
     private HashSet<Interactable> interactables = new HashSet<Interactable>(); // List of interactables in range
-
+
```
Comment (ddabble) [inline Assets/Scripts/Player/PlayerStateController.cs:33]: These spaces (and the ones below) should maybe also be removed 😅

File: Assets/Scripts/Player/PlayerStateController.cs [hunk 2]
```diff
@@ -204,11 +199,7 @@ public void SetState(PlayerStates state)
                 if (heldInteractable)
                 {
                     //Refund turret
-                    inventoryManager.ResourceAmount += ui.GetSelectedSegment().cost;
-
-                    RemoveInteractable(heldInteractable);
-                    Destroy(heldInteractable.gameObject);
-                    heldInteractable = null;
```
Comment (ddabble) [inline Assets/Scripts/Player/PlayerStateController.cs:211]: I don't see this line (`heldInteractable = null;`) in `SellTower()` below, is this on purpose..? 🤔

File: Assets/Scripts/ScriptableObjects/TowerScriptableObject.cs [hunk 3]
```diff
@@ -19,4 +19,9 @@ public TurretPrefabConstruction InstantiateConstructionTower(PlayerStateControll
         controller.PrepareTurret(spawnedConstructionTower.GetComponent<Interactable>());
         return spawnedConstructionTower.GetComponent<TurretPrefabConstruction>();
     }
+
+    public int getCost()
```
Comment (ddabble) [inline Assets/Scripts/ScriptableObjects/TowerScriptableObject.cs:23]:
```suggestion
    public int GetCost()
```
😋

Also, as the `cost` field is already public, should we either set it to private (with `[SerializeField]`), or just remove this method altogether? 🤔

File: Assets/Scripts/Player/PlayerStateController.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Player/PlayerStateController.cs:211]: Indeed it is. It turns out, `heldInteractable` is and has been broken for a long time. Nothing except loot currently uses it, and carrying loot has been prone to bugs in playtesting, most likely due to this. I have already created a Trello card in the backlog when I discovered this while making `SellTower().`

Speaking of `SellTower()`, it should function just fine without `heldInteractable = null;`, as it is completly independent of that variable as of this pull request. 😎

File: Assets/Scripts/ScriptableObjects/TowerScriptableObject.cs (hunk 3 above)
Comment (Fueredoriku) [reply Assets/Scripts/ScriptableObjects/TowerScriptableObject.cs:23]: `cost` should be private, as it makes `SellTower()` read better IMO. Will push a commit adressing this asap

**Review by ddabble — APPROVED** [review]
Niice 😄

**Pull request merged**